File: backend/src/main/java/com/sunline/sunline_backend/service/OrderService.java
package com.sunline.sunline_backend.service;

import com.sunline.sunline_backend.dto.OrderDto;
import com.sunline.sunline_backend.dto.PlaceOrderRequest;
import com.sunline.sunline_backend.entity.OrderStatus;
import java.util.List;

public interface OrderService {
    OrderDto placeOrder(String userEmail, PlaceOrderRequest request);
    List<OrderDto> getMyOrders(String userEmail);
    OrderDto cancelOrder(Long orderId, String userEmail);
    OrderDto updateOrderStatus(Long orderId, OrderStatus status);
    List<OrderDto> getAllOrders();
    List<OrderDto> getKitchenOrders();
    List<OrderDto> getDeliveryOrders();
    long countOrdersByStatus(OrderStatus status);
    long countAllOrders();
}

File: frontend/src/services/orderService.js
import api from './api';

const orderService = {
    placeOrder: async (orderData) => {
        const response = await api.post('/orders', orderData);
        return response.data;
    },

    getMyOrders: async () => {
        const response = await api.get('/orders/my-orders');
        return response.data;
    },

    cancelOrder: async (orderId) => {
        const response = await api.put(`/orders/${orderId}/cancel`);
        return response.data;
    },

    updateOrderStatus: async (orderId, status) => {
        const response = await api.patch(`/orders/${orderId}/status`, { status });
        return response.data;
    },

    getKitchenOrders: async () => {
        const response = await api.get('/orders/kitchen');
        return response.data;
    },

    getDeliveryOrders: async () => {
        const response = await api.get('/orders/delivery');
        return response.data;
    },

    getAllOrders: async () => {
        const response = await api.get('/orders/all');
        return response.data;
    }
};

export default orderService;
